pkg/pool-weighted/contracts/managed/ManagedPoolTokenLib.sol

```diff
@@ -39,27 +38,25 @@ import "../lib/ValueCompression.sol";
 library ManagedPoolTokenLib {
     using WordCodec for bytes32;
     using FixedPoint for uint256;
-    using ValueCompression for uint256;
 
     // Store token-based values:
     // Each token's scaling factor (encoded as the scaling factor's exponent / token decimals).
     // Each token's starting and ending denormalized weights.
-    // [ 123 bits |  5 bits  |       64 bits     |       64 bits       |
+    // [ 119 bits |  5 bits  |       66 bits     |       66 bits       |
     // [  unused  | decimals | end denorm weight | start denorm weight |
     // |MSB                                                         LSB|
     uint256 private constant _START_DENORM_WEIGHT_OFFSET = 0;
     uint256 private constant _END_DENORM_WEIGHT_OFFSET = _START_DENORM_WEIGHT_OFFSET + _DENORM_WEIGHT_WIDTH;
     uint256 private constant _DECIMAL_DIFF_OFFSET = _END_DENORM_WEIGHT_OFFSET + _DENORM_WEIGHT_WIDTH;
 
-    uint256 private constant _DENORM_WEIGHT_WIDTH = 64;
+    // The maximum allowable value for `denormWeightSum` is `50e18`, produced by the situation where a token is added
+    // to a Pool with equal weights such that all the other tokens are reduced to the minimum weight.
+    // The maximum denormalized weight of a single token is then given by the 3 token case of a 1%:1%:98% weight Pool
+    // where the 98% token has a denormalized weight of `49e18`.
+    // A 66 bit value allows storing values up to ~73e18 and so is suitable for storing all potential token weights.
+    uint256 private constant _DENORM_WEIGHT_WIDTH = 66;
```

Comment:
I don't think this is accurate. E.g. weights are 50/50. Add two tokens at 49, they become 1/1/49/49. The sum is `0.5+0.5+0.5 *49 + 0.5 * 49 = 50`. I then remove the two tokens at 1%, the weights are 50/50 and the sum is 49. Repeat ad infinitum.